### ISSUE TITLE
Id is abscent in payload on saving a new model

### ISSUE THREAD

**Issue by novtor** [emberjs/data#3360]
When I create a model
```javascript
var newPost = this.store.createRecord('post', {
'id': '',
'name':'good post'
});
newPost.set('id', 'qwerty');
newPst.save();
```
the id 'qwerty' is abscent in the server call.
Is it expected behavior?

**Comment by btecu** [comment]
You don't have to pass: `'id': ''` when you create the record.
Also, it may just be a typo, but I see you're calling `save` on `newPst` instead of `newPost` (maybe you have multiple new records?).

**Comment by bmac** [comment]
Hi @novtor, that does not seem like expected behavior. Would you be able to make a jsbin demonstrating the issue? Here is a jsbin to get you started http://emberjs.jsbin.com/navucuvoso/1/edit?html,js

**Comment by novtor** [comment]
Hi @btecu, @bmac. Thank you for your replies. In my usecase I need to precise the id of the record being saved so well for the moment I made a workaround: I pass the id in another field and the server takes it as id and real 'id' field is ignored by the server.
Here is the example:
http://emberjs.jsbin.com/qaqiyi/edit?html,js,output
Thank you @bmac for the sample. Click on the 'new author' and enter an id and a name, on 'save' click it tries to save this entity and the payload is shown in the alert popup (except that firefox tells me that window.alert is not a function-((( but chrome do the job).

**Comment by sly7-7** [comment]
@novtor for creating a model with a primaryKey, I think you must pass the id in the `createRecord`data hash. I don't think you can modify the primaryKey after a record is created. At least, internally, only the original id is put in the internalModel, and looks immutable. I don't know if we want it to track the record's id change.
cc/ @igorT @wecc

**Comment by novtor** [comment]
Hi @sly7-7 thank you for your reply. So it looks to be expected.
And just realised I forgot to precise that it used to work fine before 1.0.0-beta.19.

**Comment by sly7-7** [comment]
Hmm, so if it used to work pre beta.19, that should mean it is maybe something we forgot when introducing internalModels.

**Comment by bierdybard** [comment]
+1, I am currently experiencing this issue as well, starting with ED 1.0.0b-beta.19.

**Comment by bierdybard** [comment]
@sly7-7 It would seem to me that having the id be immutable after the record is committed would be appropriate, but before committing I think we should be able to change it.

**Comment by sly7-7** [comment]
@wecc Do you have any thoughts on this ?

**Comment by wecc** [comment]
No, need to poke @igorT brain to understand Modal/InternalModel better.

**Comment by jgelens** [comment]
I can confirm this issue too. Imo the id should be immutable after saving (POST) for the first time. But right after createRecord it still should be callable.

**Comment by wecc** [comment]
Closing as I'm pretty sure this have been fixed by #3701, please re-open if I'm mistaken.
